tests/test_algorithm.py

```diff
@@ -706,7 +706,7 @@ def setUpClass(cls):
         for sid in cls.sids:
             equities_metadata[sid] = {
                 'start_date': cls.sim_params.period_start,
-                'end_date': cls.sim_params.period_end
+                'end_date': cls.env.next_trading_day(cls.sim_params.period_end)
```

Comment:
because the mock data is written up to, and including, the last day of the trading range.  the asset's end date is the first day for which we don't have data.

the change to `price` to return `NaN` after the asset end date exposed this mismatch.